integration-tests/src/test/java/org/apache/druid/tests/coordinator/duty/ITAutoCompactionTest.java

```diff
@@ -636,10 +682,11 @@ public void testAutoCompactionDutyCanUpdateCompactionConfig(CompactionEngine eng
 
         final HashedPartitionsSpec hashedPartitionsSpec = new HashedPartitionsSpec(null, 3, null);
         submitCompactionConfig(hashedPartitionsSpec, NO_SKIP_OFFSET, 1, null, null, null, null, false, engine);
-        // 2 segments published per day after compaction.
-        forceTriggerAutoCompaction(4);
+        // 2 and 3 segments resp for 2013-08-31 and 2013-09-01, resp.
+        // (numShards guarantees max shards but not exact number of final shards, since some shards may end up empty.)
+        forceTriggerAutoCompaction(5);
         verifyQuery(INDEX_QUERIES_RESOURCE);
-        verifySegmentsCompacted(hashedPartitionsSpec, 4);
+        verifySegmentsCompacted(hashedPartitionsSpec, 5);
```

Comment:
Why do we need to change this test?

Reply:
Have a comment a few lines above. Reqd since numShards is a (max) hint and the actual number can change based on the data.